msbuild/Xamarin.MacDev.Tasks/Decompress.cs

```diff
@@ -0,0 +1,220 @@
+				if (contentEntry is null) {
+					log.LogWarning (MSBStrings.W7106 /* Expected a file named '{1}' in the zip file {0}. */, resources, relativeFilePath);
+					return null;
+				}
+
+				using var contentStream = contentEntry.Open ();
+				var memoryStream = new MemoryStream ();
+				contentStream.CopyTo (memoryStream);
+				memoryStream.Position = 0;
+				return memoryStream;
+			}
+
+			if (!Directory.Exists (resources)) {
+				log.LogWarning (MSBStrings.W7111 /* The directory '{0}' does not exist. */, resources);
+				return null;
+			}
+
+			var contentPath = Path.Combine (resources, relativeFilePath);
+			if (!File.Exists (contentPath)) {
+				log.LogWarning (MSBStrings.W7108 /* The file '{0}' does not exist. */, contentPath);
+				return null;
+			}
+
+			return File.OpenRead (contentPath);
+		}
+
+		/// <summary>
+		/// Extracts the specified resource (may be either a file or a directory) from the given zip file.
+		/// A stamp file will be created to avoid re-extracting unnecessarily.
+		///
+		/// Fails if:
+		/// * The resource is or contains a symlink and we're executing on Windows.
+		/// * The resource isn't found inside the zip file.
+		/// </summary>
+		/// <param name="log"></param>
+		/// <param name="zip">The zip to search in</param>
+		/// <param name="resource">The relative path inside the zip to extract (may be a file or a directory).</param>
+		/// <param name="decompressionDir">The location on disk to store the extracted results</param>
+		/// <param name="decompressedResource">The location on disk to the extracted resource</param>
+		/// <returns></returns>
+		public static bool TryDecompress (TaskLoggingHelper log, string zip, string resource, string decompressionDir, List<string> createdFiles, [NotNullWhen (true)] out string? decompressedResource)
+		{
+			decompressedResource = Path.Combine (decompressionDir, resource);
+
+			var stampFile = decompressedResource + ".stamp";
+
+			if (FileCopier.IsUptodate (zip, stampFile, XamarinTask.GetFileCopierReportErrorCallback (log), XamarinTask.GetFileCopierLogCallback (log), check_stamp: false))
+				return true;
+
+			// We use 'unzip' to extract on !Windows, and System.IO.Compression to extract on Windows.
+			// This is because System.IO.Compression doesn't handle symlinks correctly, so we can only use
+			// it on Windows. It's also possible to set the XAMARIN_USE_SYSTEM_IO_COMPRESSION=1 environment
+			// variable to force using System.IO.Compression on !Windows, which is particularly useful when
+			// testing the System.IO.Compression implementation locally (with the caveat that symlinks won't
+			// be extracted).
+
+			bool rv;
+			if (Environment.OSVersion.Platform == PlatformID.Win32NT) {
+				rv = TryDecompressUsingSystemIOCompression (log, zip, resource, decompressionDir);
+			} else if (!string.IsNullOrEmpty (Environment.GetEnvironmentVariable ("XAMARIN_USE_SYSTEM_IO_COMPRESSION"))) {
```

Comment:
Shouldn't this be part of the previous if statement using an `or` operator instead of in an `else if` statement since both execute the exact same line?